server/src/main/java/io/crate/planner/operators/InsertFromValues.java

```diff
@@ -254,24 +257,28 @@ public void execute(DependencyCarrier dependencies,
             false);
 
         var shardedRequests = new ShardedRequests<>(builder::newRequest, RamAccounting.NO_ACCOUNTING);
-        HashMap<String, Consumer<IndexItem>> validatorsCache = new HashMap<>();
         for (Row row : rows) {
             Item item = grouper.apply(shardedRequests, row, true);
 
             try {
                 // Primary Key and CLUSTERED BY check is already done in grouper -> RowShardResolver, both cannot be null.
+                // checkConstraintsOnGeneratedSource is also done in grouper.apply but only for partitioned tables.
+                // It's important to keep check below so that
+                // insert from values into regular tables doesn't leave partitions behind in case of validation failure.
```

Comment:
The comment is confusing as on regular tables (non-partitioned), no bad partition exists without proper validation.
I think the validation here is done for optimization reasons as all values already exists and thus we avoid sending invalid ones to the shards?

Reply:
thanks, comment was totally wrong. 
What I had in mind was keeping old execution logic for Insert from values into regular tables.

Indeed, abandoned partitions issue is not related to this case.

I think we can even drop this check now, wdyt? 

For now I kept it and updated the comment that we just want to fail early and don't send invalid values to the shards.